src/modals/add-liquidity-modal/add-liquidity-modal.tsx

```diff
@@ -0,0 +1,206 @@
+import { formatSize } from '../../styles/format-size';
+import { LIQUIDITY_BAKING_DEX_ADDRESS, TZ_BTC_TOKEN_SLUG } from '../../token/data/token-slugs';
+import { emptyToken } from '../../token/interfaces/token.interface';
+import { getTokenSlug } from '../../token/utils/token.utils';
+import { findExchangeRate, findLpTokenAmount, findTokenInput } from '../../utils/dex.utils';
+import { isDefined } from '../../utils/is-defined';
+import { formatAssetAmount } from '../../utils/number.util';
+import { mutezToTz, tzToMutez } from '../../utils/tezos.util';
+import { parseTransferParamsToParamsWithKind } from '../../utils/transfer-params.utils';
+import { addLiquidityModalValidationSchema, AddLiquidityModalFormValues } from './add-liquidity-modal.form';
+import { useAddLiquidityModalStyles } from './add-liquidity-modal.styles';
+
+export const AddLiquidityModal = () => {
+  const { navigate } = useNavigation();
+  const styles = useAddLiquidityModalStyles();
+  const assetsList = useAssetsListSelector();
+  const { publicKeyHash } = useSelectedAccountSelector();
+  const [tzBtcAddress] = TZ_BTC_TOKEN_SLUG.split('_');
+
+  const aToken = useTezosTokenSelector() ?? emptyToken;
+  const bToken = assetsList.find(token => getTokenSlug(token) === TZ_BTC_TOKEN_SLUG) ?? emptyToken;
+
+  const { storage, contract } = useContract<LiquidityBakingContractAbstraction, LiquidityBakingStorage>(
+    LIQUIDITY_BAKING_DEX_ADDRESS,
+    liquidityBakingStorageInitialValue
+  );
+  const aTokenPool = storage.xtzPool;
+  const bTokenPool = storage.tokenPool;
+  const lpTotalSupply = storage.lqtTotal;
+
+  const { contract: tzBtcTokenContract } = useContract(tzBtcAddress, undefined);
+
+  const onSubmitHandler = (values: AddLiquidityModalFormValues) => {
+    if (
+      isDefined(values.aToken.amount) &&
+      isDefined(values.bToken.amount) &&
+      isDefined(tzBtcTokenContract) &&
+      isDefined(contract)
+    ) {
+      const lpTokensOutput = findLpTokenAmount(values.aToken.amount, lpTotalSupply, aTokenPool);
+
+      const zeroApproveOpParams = tzBtcTokenContract.methods
+        .approve(LIQUIDITY_BAKING_DEX_ADDRESS, new BigNumber(0))
+        .toTransferParams({ mutez: true });
+
+      const bTokenApproveOpParams = tzBtcTokenContract.methods
+        .approve(LIQUIDITY_BAKING_DEX_ADDRESS, values.bToken.amount)
+        .toTransferParams({ mutez: true });
+
+      const transferParams = contract.methods
+        .addLiquidity(publicKeyHash, lpTokensOutput, values.bToken.amount, getTransactionTimeoutDate())
+        .toTransferParams({ mutez: true, amount: values.aToken.amount.toNumber() });
+
+      const addLiquidityOpParams = parseTransferParamsToParamsWithKind(transferParams);
+
+      const opParams = [
+        ...parseTransferParamsToParamsWithKind(zeroApproveOpParams),
+        ...parseTransferParamsToParamsWithKind(bTokenApproveOpParams),
+        ...addLiquidityOpParams,
+        ...parseTransferParamsToParamsWithKind(zeroApproveOpParams)
```

Comment:
why `addLiquidityOpParams` is created if you do not do this with other TransferParams?